pkg/apis/servicecatalog/types.go

```diff
@@ -144,11 +144,13 @@ type CommonServiceBrokerSpec struct {
 //
 // ServiceClass allowed property names:
 //   name - the value set to [Cluster]ServiceClass.Name
-//   externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
-//
+//   spec.externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServiceClass.Spec.ExternalID
 // ServicePlan allowed property names:
-//   name - the value set to [Cluster]ServiceClass.Name
-//   externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   name - the value set to [Cluster]ServicePlan.Name
+//   spec.externalName - the value set to [Cluster]ServicePlan.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServicePlan.Spec.ExternalID
+//   spec.serviceClassName - the value set to [Cluster]ServicePlan.Spec.ServiceClassRef.Name
```

Comment:
`spec.serviceClassName` is a special case, it has to be `spec.serviceClassName` for ServicePlan and `spec.clusterServiceClassName` for ClusterServicePlan

https://github.com/kubernetes-incubator/service-catalog/blob/491f227a4fc7453917e347ccc957da9a312a4774/pkg/apis/servicecatalog/v1beta1/types.go#L1307

Reply:
wops I accidentally closed this... I was trying to rename my branch